core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -5054,6 +5078,28 @@ private static void computeVersion(ServletContext context) {
      * @see #getRawBuildsDir()
      */
     private static final String DEFAULT_BUILDS_DIR = "${ITEM_ROOTDIR}/builds";
+    /**
+     * Old layout for workspaces.
+     * @see #DEFAULT_WORKSPACES_DIR
+     */
+    private static final String OLD_DEFAULT_WORKSPACES_DIR = "${ITEM_ROOTDIR}/" + WORKSPACE_DIRNAME;
+
+    /**
+     * Default value for the workspace's directories layout.
+     */
+    private static final String DEFAULT_WORKSPACES_DIR = "${JENKINS_HOME}/workspace/${ITEM_FULL_NAME}";
+
+    /**
+     * System property name to set {@link #buildsDir}.
+     * @see #getRawBuildsDir()
+     */
+    public static final String BUILDS_DIR_PROP = Jenkins.class.getName() + ".BUILDS_DIR";
```

Comment:
I do not think these constants need to be `public`—we do not expect plugin Java code to refer to them.

Reply:
fixed